### PR TITLE
Unleash test of existential inference

### PR DESCRIPTION
**Pull request by som-snytt**
It doesn't entirely work, so perhaps the test is noise.

But the test/pending bot would enable the test and append the current state of failing part.

### CODE REVIEW

**Comment by joroKr21** [comment]
```scala
  def diff1[A](xs: Set[A], ys: Set[_ <: A]) = xs -- ys
  def diff2[A](xs: Set[_ >: A], ys: Set[A]) = xs -- ys
  def diff3[A](xs: Set[_ >: A], ys: Set[_ <: A]) = xs -- ys
  val xs = Set(Option(42))
  val ys = Set(Some(42))
  diff1(xs, ys)
  diff2(xs, ys) // kek
  diff3(xs, ys)
```

File: test/files/pos/existentials.scala [hunk 1]
```diff
@@ -6,17 +8,44 @@ class A {
...
-  // val bippy3 = {
-  //   lazy val bippy3a = f _
-  //   bippy3a
-  // }
+  lazy val bippy2 = f _
+  /*
+  val bippy3 = {
+    lazy val bippy3a = f _
+    bippy3a
+  }
```
Comment (hrhino) [inline test/files/pos/existentials.scala:25]: `bippy3` compiles just fine in 2.13.0-M3; better sneak the test case in before it regresses.

```scala
Welcome to Scala 2.13.0-M3 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_121).
Type in expressions for evaluation. Or try :help.

scala> def f() = { case class Bob(); Bob }
f: ()Bob.type forSome { val Bob: scala.runtime.AbstractFunction0[Bob] with Serializable{case def unapply(x$0: Bob): Boolean}; type Bob <: Product with Serializable{def copy(): Bob} }

scala>

scala> :pa
// Entering paste mode (ctrl-D to finish)

  val bippy3 = {
    lazy val bippy3a = f _
    bippy3a
  }


// Exiting paste mode, now interpreting.

bippy3: Bob.type forSome { val Bob: scala.runtime.AbstractFunction0[Bob] with Serializable{case def unapply(x$0: Bob): Boolean}; type Bob <: Product with Serializable{def copy(): Bob} } = Bob

scala>
```

Comment (lrytz) [reply test/files/pos/existentials.scala:25]: we can un-comment it in 2.13.x after this PR has been merged forward.

**Review by lrytz — APPROVED** [review]

File: test/files/pos/existentials.scala (hunk 1 above)
Comment (hrhino) [reply test/files/pos/existentials.scala:25]: Yep, that was what I meant. 👍

**Pull request merged**